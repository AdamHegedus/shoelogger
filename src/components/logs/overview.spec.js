import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Overview from '@/components/logs/overview';

jest.mock('../charts/bar', () => {
    return {
        render: () => {
            return jest.fn().mockImplementation(() => {
                return <div class="mock-bar-chart"/>;
            });
        }
    };
});

const localVue = createLocalVue();
localVue.use(Vuex);

describe('overview.vue', () => {
    describe('when has shoes', () => {
        it('should render component', () => {
            // ASSIGN
            const navigationAction = {
                setNavigation: jest.fn()
            };

            const shoesState = {
                shoes: [0, 1, 2, 3]
            };

            const store = new Vuex.Store({
                modules: {
                    navigation: {
                        namespaced: true,
                        actions: navigationAction
                    },
                    shoes: {
                        namespaced: true,
                        state: shoesState
                    }
                }
            });

            const wrapper = shallowMount(Overview, { store, localVue });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
            expect(navigationAction.setNavigation.mock.calls.length).toMatchSnapshot();
            expect(navigationAction.setNavigation.mock.calls[0][1]).toMatchSnapshot();
        });
    });

    describe('when doesn\'t have shoes', () => {
        it('should render component', () => {
            // ASSIGN
            const navigationAction = {
                setNavigation: jest.fn()
            };

            const shoesState = {
                shoes: []
            };

            const store = new Vuex.Store({
                modules: {
                    navigation: {
                        namespaced: true,
                        actions: navigationAction
                    },
                    shoes: {
                        namespaced: true,
                        state: shoesState
                    }
                }
            });

            const wrapper = shallowMount(Overview, { store, localVue });

            // ACT
            const actual = wrapper.element;

            // ASSERT
            expect(actual).toMatchSnapshot();
            expect(navigationAction.setNavigation.mock.calls.length).toMatchSnapshot();
            expect(navigationAction.setNavigation.mock.calls[0][1]).toMatchSnapshot();
        });
    });
});
