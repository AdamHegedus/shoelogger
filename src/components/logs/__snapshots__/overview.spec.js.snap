// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overview.vue when doesn't have shoes should render component 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h3>
        Overview
      </h3>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`overview.vue when doesn't have shoes should render component 2`] = `1`;

exports[`overview.vue when doesn't have shoes should render component 3`] = `"/logs/overview"`;

exports[`overview.vue when has shoes should render component 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm col-lg-6"
    >
      <h3>
        Overview
      </h3>
       
      <bar-stub
        data="0,1,2,3"
        style="height: 230px;"
      />
    </div>
  </div>
</div>
`;

exports[`overview.vue when has shoes should render component 2`] = `1`;

exports[`overview.vue when has shoes should render component 3`] = `"/logs/overview"`;
